indigo/indigo-extras/src/main/scala/indigoextras/waypoints/WaypointPath.scala

```diff
@@ -0,0 +1,113 @@
+package indigoextras.waypoints
+
+import indigo.Radians
+import indigo.Vector2
+
+import scala.annotation.tailrec
+
+trait WaypointPath:
+  def waypoints: List[Vector2]
+
+  def effectiveWaypoints: List[Vector2]
+
+  def loop: Boolean
+
+  def calculatePosition(
+      at: Double
+  ): (Vector2, Radians)
+
+object WaypointPath:
+  def apply(
+      waypoints: List[Vector2],
+      radius: Double,
+      loop: Boolean
+  ): WaypointPath =
```

Comment:
To keep with Indigo's API style, we could do with adding two more apply methods that call this one:

```scala
def apply(waypoints: List[Vector2]): WaypointPath
def apply(waypoints: List[Vector2], radius: Double): WaypointPath
```